tutorials/appengine-pusher/index.md

```diff
@@ -0,0 +1,450 @@
+Custom event handlers can be attached to a given event type. 
+This allows for efficient event routing in the clients.
+Note: A subscriber will receive all messages published over a channel.
+
+Events may be trigged by the user or Pusher.
+In case of Pusher-triggered events on a channel, the event name is
+prefixed with `pusher:`, such as `pusher:subscription-succeeded`.
+
+## Chat application
+The sample application demonstrates presence channels in Pusher for a chat application.
+View complete source code [here](java).
+     
+The [Java server-side SDK](https://github.com/pusher/pusher-http-java) is used for authorizing 
+Pusher subscriptions and publishing events to the channel, and the
+[JavaScript WebSocket SDK](https://github.com/pusher/pusher-js) is used to 
+subscribe to the events.
+     
+All users subscribed to the channel receive updates when users connect or disconnect from the channel.
+
+## Using Pusher on your server
+The server-side REST SDK is used to initialize a Pusher instance, 
+authorize secure presence channels, and provide clients an endpoint to trigger events.
+
+### Connecting to Pusher
+The following code provides examples of initializing and connecting to Pusher.
+Use the credentials from the application you created to initialize and connect to Pusher,
+as shown in this example.
+
+Note: It is important to provide the cluster information if not using the default `mt1` (`us-east-1`) cluster.
+You can encrypt messages sent over Pusher.
+
+[embedmd]:# (java/src/main/java/com/example/appengine/pusher/PusherService.java /public abstract/ $)
+```java
+public abstract class PusherService {
+
+  public static final String APP_KEY = System.getenv("PUSHER_APP_KEY");
+  public static final String CLUSTER = System.getenv("PUSHER_CLUSTER");
+
+  private static final String APP_ID = System.getenv("PUSHER_APP_ID");
+  private static final String APP_SECRET = System.getenv("PUSHER_APP_SECRET");
+
+  private static Pusher instance;
+
+  static Pusher getDefaultInstance() {
+    if (instance != null) {
+      return instance;
+    } // Instantiate a pusher
+    Pusher pusher = new Pusher(APP_ID, APP_KEY, APP_SECRET);
+    pusher.setCluster(CLUSTER); // required, if not default mt1 (us-east-1)
+    pusher.setEncrypted(true); // optional, ensure subscriber also matches these settings
+    instance = pusher;
+    return pusher;
+  }
+}
+```
+
+### Authorize client subscriptions for secure channels
+
+Clients connecting to private or presence channels require server-side authentication.
+The authentication endpoint can be implemented in your server as shown in this example.
+The REST SDK provides methods to retrieve the required authentication JSON.
+
+Note: Private channels do not require user information to be provided as part of the authentication.
+To learn more about authentication in Pusher, see [Pusher's documentation](https://pusher.com/docs/authenticating_users).
```

Comment:
"To learn more about authentication in Pusher, see [Pusher's documentation](https://pusher.com/docs/authenticating_users)"

Change to:

"To learn more about authentication in Pusher, refer to Pusher's [Authenticating users](https://pusher.com/docs/authenticating_users) documentation."